### PR TITLE
🎉 Source Zendesk: sync rate improvement

### LINKED ISSUE #9062: 🐛  Source Zendesk Support: sync rate improvement
*Is this your first time deploying Airbyte:* Yes
*OS Version / Instance:* MBP / Catalina
*Deployment:* docker-compose
*Airbyte Version:* 0.34.1-alpha
*Source name/version:* Zendesk Support 0.1.8
*Destination name/version:* Snowflake
*Description:* I am testing out Zendesk-&gt;Snowflake sync on my laptop before deploying to an EC2 instance. Everything is working, but the rate at which the data is syncing is very slow. As you can see in the logs, it is syncing at a rate of 1K rows every minute or so. We have &gt;7M tickets in Zendesk, so total number of table rows is some multiple of that, let’s say 50M rows, which means that at 1000 rows/minute it will take over a month to backfill our data.

I emailed with Zendesk Support about this, and they said that given our API rate limit of 700 requests per minute and the # of rows that can be pulled per request, we should be able to backfill all of our data in less than 3 days. I just checked our Zendesk API status page, which says we are at 163/700 requests per minute at this very moment (with Airbyte chugging away).

Since this is my first time using Airbyte, I’m guessing/hoping that I missed a setting somewhere. Thanks in advance for your help!

 https://airbytehq.slack.com/archives/C01MFR03D5W/p1639967338010700?thread_ts=1639967338.010700&cid=C01MFR03D5W

### PR DESCRIPTION
**Pull request by htrueman**
## What
Updated Source Zendesk request execution with future requests.

## How
Used requests-future library using FuturesSession, then processing them in batch.

## Recommended reading order
1. `streams.py`
2. `setup.py`

## 🚨 User Impact 🚨
It shouldn't change the UX.

## Pre-merge Checklist
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/9456)

### CODE REVIEW

File: airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py [hunk 1]
```diff
@@ -37,12 +60,66 @@ class SourceZendeskSupportStream(HttpStream, ABC):
...
         self._subdomain = subdomain
 
+    def _send(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+        response: requests.Response = self._session.send(request, **request_kwargs)
+        return response
+
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+        return self._send(request, request_kwargs)
+
+    def read_records(
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py:78]: this is repeating the whole CDK method, can you add some context on why? ideally this should not be the case

**Review by sergei-solonitcyn — CHANGES_REQUESTED** [review]

**Review by sergei-solonitcyn — APPROVED** [review]

**Pull request merged**